packages/metrics-util/src/nodejs/CoreDataSource.js

```diff
@@ -5,33 +5,51 @@
 
 'use strict';
 
+const core = require('@instana/core');
+const sharedMetrics = require('@instana/shared-metrics');
+const { consoleLogger } = require('@instana/serverless');
 const DataSource = require('../DataSource');
 
 /**
- * A source for snapshot data and metrics that adapts the metrics from @instana/core and @instana/shared-metrics.
+ * A source for snapshot data and metrics that adapts the metrics
+ * from @instana/core and @instana/shared-metrics.
+ *
+ * `activate` will be called from the registry.
+ * The NodejsProcessor owns the CoreDataSource.
+ *
+ * > allProcessors.forEach(processor => processor.activate());
+ *
+ * This will activate core & shared-metrics.
  */
 class CoreDataSource extends DataSource {
-  constructor(coreMetrics, refreshDelay) {
```

Comment:
This class owns, defines and holds the data for "CoreData".
As soon as you instantiate the class, you expect to get data.

```
source = new CoreDataSource({ some config })
source.activate()
source.doRefresh()
source.on('data')
```